roles/openshift_named_certificates/tasks/main.yml

```diff
@@ -40,7 +47,29 @@
 
 - name: Land named CA certificates
   copy:
-    src: "{{ item }}"
-    dest: "{{ named_certs_dir }}/{{ item | basename }}"
+    src: "{{ item.cafile }}"
+    dest: "{{ named_certs_dir }}/{{ item.cafile | basename }}"
+    mode: 0600
+  with_items: "{{ named_certificates }}"
+  when: "'cafile' in item"
+
+- name: Land named CA certificates in temporary fragment directories
+  copy:
+    src: "{{ item.cafile }}"
+    dest: "{{ named_certs_dir }}/fragments/{{ item.certfile | basename }}/01_{{ item.cafile | basename }}"
+    mode: 0600
+  with_items: "{{ named_certificates }}"
+  when: "'cafile' in item"
+
+- name: Compile cert files from fragments
+  assemble:
+    src: "{{ named_certs_dir }}/fragments/{{ item.certfile | basename }}"
+    dest: "{{ named_certs_dir }}/{{ item.certfile | basename }}"
     mode: 0600
-  with_items: "{{ named_certificates | oo_collect('cafile') }}"
+    validate: "openssl verify %s"
```

Comment:
I think this validate is too strict and would only work if we already trusted the root? Using something like [this guide](https://jamielinux.com/docs/openssl-certificate-authority/index.html) to roll your own chain would be a valid configuration for a cluster but would require that clients trust the root.

Reply:
The reason i went this route was that ither the CAfile provided is a fully trusted bundle or the system trusts the CA already. Otherwise you arent necessarily working with a working cert chain.